app.py

```diff
@@ -7,119 +7,129 @@
+        user = User.query.filter_by(username=username).first()
+        if user and bcrypt.check_password_hash(user.password, password):
+            session['user_id'] = userid
+            session['username'] = username
+            return redirect(url_for('dashboard'))
+        flash('The password is incorrect!')
+        return redirect(url_for('home'))
+
+    return render_template("login.html")
+
+
+@app.route('/dashboard', methods=['GET', 'POST'])
 def dashboard():
-	start_date = '2022-07-01'
-	end_date = 'today'
-	if 'user_id' in session:
-		user_id = session['user_id']
-		username = 'karen'
-		if request.method =='POST':
-			start_date = request.form['start_date']
-			end_date = request.form['end_date']
-			
-			if start_date > end_date:
-				flash('Please enter correct dates')
-				start_date = '2022-07-01'
-				end_date = 'today'
-			if start_date=='' or end_date == '' :
-				flash('Please enter a start date and end date')
-				start_date = '2022-07-01'
-				end_date = 'today'
-
-			if start_date < '2022-07-11':
-				flash('please enter a start date after July 11th')
-				start_date = '2022-07-01'
-				end_date = 'today'
-
-			if start_date > str(date.today()):
-				flash('You cannot enter a date in the future')
-				start_date = '2022-07-01'
-				end_date = 'today'
-
-			
-			
-
-		return render_template("dashboard.html", username=username, terms_dict= ga.get_terms(), analytics_dict=ga.get_analytics_date(start_date,end_date))
-	else:
-		return redirect(url_for('home')) 
-	return render_template('dashboard.html', username=username)
-
-@app.route('/search_terms', methods=['GET','POST'])	
+    # Start
+    start_date = '2022-07-01'
+    end_date = 'today'
+    if 'user_id' in session:
+        user_id = session['user_id']
+        username = session['username']
+        if request.method == 'POST':
+            start_date = request.form['start_date']
+            end_date = request.form['end_date']
+
+            correct_date = date_check(start_date, end_date)
+            start_date = correct_date['startdate']
+            end_date = correct_date['enddate']
+            if correct_date['message'] != '':
+                start_date = '2022-07-01'
+                end_date = 'today'
+                flash(correct_date['message'])
+
+        return render_template("dashboard.html", username=username, terms_dict=ga.get_terms(), analytics_dict=ga.get_analytics_date(start_date, end_date))
+    else:
+        return redirect(url_for('home'))
+    return render_template('dashboard.html', username=username)
+
+
+@app.route('/search_terms', methods=['GET', 'POST'])
 def search_terms():
-	start_date = '2022-07-01'
-	end_date = 'today'
-	if request.method == 'POST':
-		start_date = request.form['start_date']
-		end_date = request.form['end_date']
-	return render_template('terms.html', terms_dict= ga.get_analytics_date(start_date,end_date), data=ga.get_analytics_date(start_date,end_date)['search_data'])
+    start_date = '2022-07-01'
```

Comment:
Whew, I'm really sorry. This whole PR slipped my mind.

I see that this date is 2022-07-01, while the date_check is checking for dates after 2022-07-11. Which date is supposed to be the minimum date? I think they should all match, right?

You may just want to declare a constant MIN_DATE at the top of this file and use it in the place of literals, just to make sure it's always consistent.

Reply:
Oh That makes sense!